### DISCUSSION TITLE
dupe search

### DISCUSSION THREAD

**Discussion by klapvogn** [ryelogheat/xpbot#48] [General]
What am i doing wrong here:
[https://prnt.sc/26wdgwel](https://prnt.sc/26wdgwe)

**Comment by ryelogheat** [comment]
The search response was not returned in an expected format. The way to fix it is going to require you to modify this bit of JSON https://github.com/ryelogheat/xpbot/blob/de607ea23dd42d8329d1337b79eda052b17b859c/site_templates/blutopia.json#L139-L146 (Obviously replace **blutopia.json** with whatever site you're actually uploading to)

To figure out the correct keys/format you need to try making a search query request manually and analyze the result, the URL for that on Blutopia looks something like this:
https://blutopia.xyz/api/torrents/filter?api_token=XXXXXXXXXXXXX&imdbId=13668894 
![json-resp](https://i.ibb.co/3hz70J2/json-resp.jpg)

In that ^^ screenshot you can see the **top level** object name is `data` then to get the actual torrent details you need to look under `attributes` (**Note**: The default configuration clearly isn't working for whatever site you're uploading to so you need to substitute **data** & **attributes** with whatever you find when you make that ^^ request on your site)

So if you look at **blutopia.json** again you'll see that in this line https://github.com/ryelogheat/xpbot/blob/de607ea23dd42d8329d1337b79eda052b17b859c/site_templates/blutopia.json#L144-L145 we specify `data` and right below that we specify `attributes`


tldr; you need to fix your **tracker.json** file so it uses the correct JSON objects/keys when parsing the response

**Comment by klapvogn** [comment]
Thanks for the reply, 

one question, how do i get to the image you are showing so i can see what i have?

**Comment by ryelogheat** [comment]
Replace **blutopia.xyz** with the tracker you're uploading to in this URL https://blutopia.xyz/api/torrents/filter?api_token=XXXXXXXXXXXXX&imdbId=13668894 (also input your actual API Key instead of the X's) 

Then just take that whole URL ^^ and open it in a new tab

**Reply by klapvogn** [reply]
So fare so good, did that :)

**Reply by ryelogheat** [reply]
What did you find? What was the top level object name? You can just post a screenshot if you're not sure

**Comment by klapvogn** [comment]
found this: https://prnt.sc/26wf68k

**Reply by ryelogheat** [reply]
OK so that's kinda weird, your site returns results similar to what BHD does instead of vanilla UNIT3D but that aside all you need to do is take these lines in your **tracker.json** file
```
"parse_json": {
  "top_lvl": "data",
  "torrent_details": "attributes"
},
```
and replace them with the following
```
"parse_json": {
  "top_lvl": "torrents"
},
```

**Comment by klapvogn** [comment]
Something worked \o/ but it gave and error:
`Site response: {"success":false,"message":"Validation Error","data":"A .NFO file need to be uploaded."}`

**Reply by ryelogheat** [reply]
lol what site are you uploading to? :thinking: I can't think of any site anywhere that requires a **.nfo** file

**Reply by klapvogn** [reply]
it's a danish site `https://danishbytes.club/`

**Reply by ryelogheat** [reply]
Most P2P groups don't include **.nfo** files anymore so you'd probably have to generate your own on the fly but again I've never come across a site with mandatory **.nfo** files so I don't support generating them if they're missing

**Reply by klapvogn** [reply]
Okay thanks for the help so fare :)

**Comment by ryelogheat** [comment]
I've historically have had pretty bad luck with .nfo file uploads https://github.com/ryelogheat/xpbot/blob/de607ea23dd42d8329d1337b79eda052b17b859c/auto_upload.py#L1414-L1417 so they've been disabled indefinitely. You could try un-commenting those lines ^^ and see if it works (Note; you'll need a .nfo file in the folder that you're uploading or manually specify the path of a .nfo file with the **-nfo** arg when running the script)

Honestly I'm still a bit confused on what exactly danishbytes needs a .nfo file for & what they need it to contain

**Comment by ryelogheat** [comment]
If you do manage to get the script uploading to danishbytes but need to constantly supply a .nfo file manually then let me know & I could add a quick `if tracker == danishbytes` statement  to automatically create a .nfo file for you. 

(again if you're going to try and get danishbytes to work & they really require a .nfo file then you're going to need to uncomment these lines https://github.com/ryelogheat/xpbot/blob/de607ea23dd42d8329d1337b79eda052b17b859c/auto_upload.py#L1414-L1417 so the .nfo gets added to the upload POST request

**Comment by klapvogn** [comment]
Hey again!
this way already uncomment in the script the part about .nfo
--------------------------------------------------------------------

If the .nfo file is already present in the folder like here:
[https://prnt.sc/26x22ud](https://prnt.sc/26x22ud)

would the run code then be : `python3 auto_upload.py -t dby -anon -p /home/xxx/data/local/downloads/TORRENTLEECH/DBY/Jojo.Rabbit.2019.NORDiC.1080p.BluRay.x264-CDB/Jojo.Rabbit.2019.NORDiC.1080p.BluRay.x264-CDB.mkv`

i think i need to make you aware of that the folder also container .srt (sub) files would that cause problems?

**Reply by ryelogheat** [reply]
So in this case you'd want to upload the folder `Jojo.Rabbit.2019.NORDiC.1080p.BluRay.x264-CDB/` since it contains a few different files that all apart of your torrent (video, subs, nfo) so you need to specify the folder you want to upload not the individual .mkv file 

So your command might look something like this:
`python3 auto_upload.py -t dby -anon -p /home/xxx/data/local/downloads/TORRENTLEECH/DBY/Jojo.Rabbit.2019.NORDiC.1080p.BluRay.x264-CDB/`

The script will automatically search through the folder `Jojo.Rabbit.2019.NORDiC.1080p.BluRay.x264-CDB/` and find the **mkv** file which it uses for things like mediainfo, screenshots, etc. but it will end up uploading the folder you specified with everything in it.

**Comment by klapvogn** [comment]
I'll give it a try right now 👍 

It did find the .nfo file, but failed on upload 
[https://prnt.sc/26xulvx](https://prnt.sc/26xulvx)
^ Still the error about missing .nfo 

with this code: `sudo python3 auto_upload.py -t dby -anon -p /home/xxx/data/local/downloads/TORRENTLEECH/DBY/Jojo.Rabbit.2019.NORDiC.1080p.BluRay.x264-CDB/`

**Reply by klapvogn** [reply]
got i working with nfo \o/

**Reply by ryelogheat** [reply]
How did you get it working? Do you still need a .nfo file when uploading?

**Reply by klapvogn** [reply]
yes still need the file

under `Required`  in the `xxx.json` file
i had `"nfo_file": "file"` but it should be : `"nfo": "file"`

**Reply by klapvogn** [reply]
all working now, but one issue when it's upload i get
`Tracker Message | Tracker:  [Could not parse bencoded data: Error 404: Page Not Found  - DBy v0.95 `       

do the script download the torrent file from the site or use the one it creates when uploading?

**Reply by ryelogheat** [reply]
That's right! I forgot you can change that, there's some sites like BHD that require you to open the .nfo file and upload the text and others had different keys for the json payload, etc.

**Comment by ryelogheat** [comment]
> all working now, but one issue when it's upload i get `Tracker Message | Tracker: [Could not parse bencoded data: Error 404: Page Not Found - DBy v0.95 `
> 
> do the script download the torrent file from the site or use the one it creates when uploading?

It uses the .torrent file it generated, you should have provided the trackers "Source" in **tracker.json** which should let you start seeding with the generated file

for example:
![image](https://img.belko.pw/7d/8ef1bc3baee34e3be6a6859aff5007a4066c4e.png)
https://github.com/ryelogheat/xpbot/blob/de607ea23dd42d8329d1337b79eda052b17b859c/site_templates/blutopia.json#L6

**Reply by klapvogn** [reply]
Don't follow?

**Reply by ryelogheat** [reply]
In **<your_tracker>.json** what does it say on line 6?

**Reply by klapvogn** [reply]
i have `"source": "dby",` so it needs to be `"source": "blu",` ?

**Reply by ryelogheat** [reply]
Is **dby** correct? Is that what danishbytes says to use? I used **blu** as an example since on the Blutopia upload form they show that message about the source. Some sites might have you use the trackers full name as the source so you need to find out what the correct source is

**Reply by klapvogn** [reply]
Hmm how can i find out what source it is, i know its unit3d?

**Reply by ryelogheat** [reply]
Go to the sites upload page and look for it there

**Reply by klapvogn** [reply]
hmm in the sourcecode of the upload page?

**Reply by ryelogheat** [reply]
No, it should literally be right on the upload page
![image](https://img.belko.pw/91/158f4be69bcf967f6c5fedec36488914e37a2b.jpg)

**Reply by klapvogn** [reply]
I see: private flag and source to DANISH BYTES

**Reply by ryelogheat** [reply]
OK good well there's your answer. You need to open up **<your_tracker>.json** and change the source to **DANISH BYTES**

**Reply by klapvogn** [reply]
Thanks a million havn't guess it my self :)

**Reply by klapvogn** [reply]
Well didn't fixed the bencoded data same error

maybe it dosn't like space between DANISH and BYTES

**Reply by ryelogheat** [reply]
OK how about this, go to danishbytes upload page and check if its asking for the nfo as a file upload or do they want you to paste text?

**Reply by ryelogheat** [reply]
Also I know that you're getting an error at the end but is the actual torrent still uploading to danishbytes? Go to danishbytes and check if your uploaded torrents are showing up

**Reply by klapvogn** [reply]
the nfo gets showned in the nfo box, when it gets the nfo file uploaded and the torrent is uploaded to the site

**Reply by ryelogheat** [reply]
So is the torrent being uploaded to danishbytes? or no

ignore everything that xpbot says right now, just check if on danishbytes.com or whatever the domain is, if your uploads are showing up

**Reply by klapvogn** [reply]
the upload is showing up on the site all good there

**Reply by ryelogheat** [reply]
OK so now that the uploads are working, check if the torrent is seeding in your torrent client. Hopefully the source update fixed the issue

**Reply by klapvogn** [reply]
dosn't seed, `Tracker: [Could not parse bencoded data: Error 404: Page Not Found`

**Comment by ryelogheat** [comment]
OK so the easiest way to fix the issue you're having is probably to download the .torrent file using the link the upload requests response returns

That probably doesn't make much sense ^^ but basically it looks like this:
**you upload** --> **server returns response** --> **link to DL .torrent in that response**

This is going to require some structural changes to **auto_upload.py** which frankly I don't want to do, if you want you can give it a shot, I'll link to some relevant functions/code that will need to be changed. 

Here is the code that calls the actual upload function. You can see that right below it is the code that moves the generated .torrent file to your watch directory, you'll need to basically remove all that and replace it with code inside of the **upload_to_site()** function that will download the .torrent file using the link you get from the site. You should probably have the downloaded .torrent file be moved inside of the **upload_to_site()** function 
  https://github.com/ryelogheat/xpbot/blob/de607ea23dd42d8329d1337b79eda052b17b859c/auto_upload.py#L1890

Again I don't really want to do any of this, I don't work on this project anymore so you either need to do it yourself or get help from someone else. Actually you should first try to figure out why the source isn't working, that might be something to try first

**Reply by klapvogn** [reply]
thanks for the writeup!

1. `"source": "<TRACKER_ACRONYM>",`
 - from `default_template.json`

2. auto_upload.py
```
# Used to correctly select json file
acronym_to_tracker = {
    "blu": "blutopia", "bhd": "beyond-hd", "r4e": "racing4everyone",
    "acm": "asiancinema", "ath": "aither", "telly": "telly",
    "ntelogo": "ntelogo", "ufhd": "uncutflixhd", "dby": "danishbytes"
}
```

i have tried many diffrent things didn't work

**Reply by klapvogn** [reply]
Lets close here.
